host/cli/discover.go

```diff
@@ -0,0 +1,190 @@
+          "FLYNN_JOB_ID": "host-b773161b-3e50-43ec-bb38-5c492c2ba2fa",
+          "FLYNN_PROCESS_TYPE": "web",
+          "FLYNN_RELEASE_ID": "305a5c6b-7f01-4232-bc1c-74b6a408960d"
+        },
+        "index": 22
+      },
+      "Instances": [
+        {
+          "id": "0bbd29d630f63b11ae999b1dabd565f9",
+          "addr": "100.100.36.4:80",
+          "proto": "http",
+          "meta": {
+            "AUTH_KEY": "9b7b3b382f13425da3a8cb390f0937b8",
+            "FLYNN_APP_ID": "2169984c-c15f-499d-be77-636612425330",
+            "FLYNN_JOB_ID": "host-b773161b-3e50-43ec-bb38-5c492c2ba2fa",
+            "FLYNN_PROCESS_TYPE": "web",
+            "FLYNN_RELEASE_ID": "305a5c6b-7f01-4232-bc1c-74b6a408960d"
+          },
+          "index": 22
+        }
+      ],
+      "Meta": {
+        "Data": {},
+        "index": 0
+      }
+    }
+  ]
+  
+  Show id and address of every 'flynn-host' service instance.
+  
+  $ flynn-host discover -f '{{range .Instances}}{{index .Meta "id"}} {{.Addr}}{{end}}' flynn-host
+  host 10.0.2.15:1113
+  
+  Show address of leader for services 'flynn-host', 'gitreceive' and 'controller'.
+  
+  $ flynn-host discover -f '{{ .Name }} {{ .Leader.Addr }}' flynn-host gitreceive controller
+  flynn-host 10.0.2.15:1113
+  gitreceive 100.100.36.8:5000
+  controller 100.100.36.4:80
+  
+  Show metadata 'primary.addr' for service 'postgres'.
+  
+  $ flynn-host discover -f '{{ metadata "primary.addr" }}' postgres
+  100.100.36.2:5432
+`)
+}
+
+type serviceInfo struct {
+	Name      string
+	Leader    *discoverd.Instance
+	Instances []*discoverd.Instance
+	Meta      serviceMeta
+}
+
+type serviceMeta struct {
+	Data  map[string]interface{}
+	Index uint64 `json:"index"`
+}
+
+func runDiscover(args *docopt.Args, c *cluster.Client) error {
+	services := args.All["<service>"].([]string)
+
+	format := args.String["--format"]
+	if args.Bool["--json"] && format != "" {
+		return errors.New("--json and --format canot be used at the same time")
+	}
+
+	if !args.Bool["--json"] && format == "" {
+		w := tabwriter.NewWriter(os.Stdout, 1, 2, 2, ' ', 0)
+		defer w.Flush()
+		listRec(w, "SERVICE", "ID", "ADDR")
+		for _, service := range services {
+			svc := discoverd.NewService(service)
+			instances, _ := svc.Instances()
+			for _, inst := range instances {
+				listRec(w, service, inst.ID, inst.Addr)
+			}
+		}
+		return nil
+	}
+
+	response := make([]serviceInfo, 0, len(services))
+
+	for _, service := range services {
+		svc := discoverd.NewService(service)
+		leader, _ := svc.Leader()
+		instances, _ := svc.Instances()
+		meta, _ := svc.GetMeta()
+
+		data := make(map[string]interface{})
+		if meta.Data != nil {
+			if err := json.Unmarshal(meta.Data, &data); err != nil {
+				return err
+			}
+		}
+
+		response = append(response, serviceInfo{
+			Name:      service,
+			Leader:    leader,
+			Instances: instances,
+			Meta: serviceMeta{
+				Index: meta.Index,
+				Data:  data,
+			},
+		})
+	}
+
+	if format == "" {
```

Comment:
Given this is checking whether JSON should be printed, it would be clearer as `if args.Bool["--json"]`